--- src/DSPack/DSUtil.pas	12 Jun 2006 17:56:11 -0000	1.23
+++ src/DSPack/DSUtil.pas	4 Apr 2007 21:18:06 -0000
@@ -2080,7 +2080,9 @@
         SysDevEnum.CreateClassEnumerator(FGUID, EnumCat, 0);
         EnumCat.Skip(index);
         EnumCat.Next(1, Moniker, nil);
-        Moniker.BindToObject(nil, nil, IID_IBaseFilter, result);
+        // Use BindMoniker function to avoid crash with some filters (e.g. Intervideo WinDVR) ...
+        if S_OK <> BindMoniker(Moniker,0, IID_IBaseFilter, result) then
+          Result := nil;
         EnumCat.Reset;
         SysDevEnum := nil;
         EnumCat    := nil;
